components/03-organisms/menu/breadcrumbs/yds-breadcrumbs.js

```diff
@@ -2,15 +2,229 @@ Drupal.behaviors.breadcrumbs = {
+    const breadcrumbsControls = context.querySelectorAll(
+      '.breadcrumbs__control',
+    );
+    const breadcrumbsLinks = context.querySelectorAll('.breadcrumbs__link');
     const breadcrumbsButton = context.querySelector('.breadcrumbs__button');
+    const breadcrumbsInner = context.querySelector('.breadcrumbs__inner');
+    const controlsWidth = context.querySelector(
+      '.breadcrumbs__control--left',
+    ).offsetWidth;
+    let scrollIndicatorDir;
 
-    // Function to add/remove breadcrumbs-overflow value.
-    // Used to show all breadcrumbs on mobile
+    /**
+     * getFirstVisible
+     * @description Get the first item that is visible (not overflown).
+     * @returns The value of the left edge of the first fully visible item plus
+     * the width of the controls so that things aren't visually hidden by the
+     * absolutely positioned elements.
+     */
+    function getFirstVisible() {
+      const breadcrumbsLeft = breadcrumbs.getBoundingClientRect().left;
+      const breadcrumbsItems =
+        breadcrumbs.querySelectorAll('.breadcrumbs__item');
+      const visibleItems = [];
+
+      breadcrumbsItems.forEach((item) => {
+        if (
+          item.getBoundingClientRect().right >
+          breadcrumbsLeft + controlsWidth
+        ) {
+          visibleItems.push(item);
+        }
+      });
+
+      return visibleItems[1].offsetLeft - controlsWidth;
+    }
+
+    /**
+     * getLastHidden
+     * @description Get the last item that is overflown (not visible).
+     * @returns The value of the left edge of the first partially hidden item
+     * minus the width of the controls so that things aren't visually hidden by
+     * the absolutely positioned elements.
+     */
+    function getLastHidden() {
+      const breadcrumbsLeft = breadcrumbs.getBoundingClientRect().left;
+      const breadcrumbsItems =
+        breadcrumbs.querySelectorAll('.breadcrumbs__item');
+      const hiddenItems = [];
+
+      breadcrumbsItems.forEach((item) => {
+        if (item.getBoundingClientRect().left < breadcrumbsLeft) {
+          hiddenItems.push(item);
+        }
+      });
+
+      return hiddenItems[hiddenItems.length - 1].offsetLeft - controlsWidth;
+    }
+
+    /**
+     * setOverflow
+     * @description Get the positions of the breadcrumbs to determine whether an
+     * overflow situation is in play.
+     */
+    function setOverflow() {
+      const breadcrumbsLeft = breadcrumbs.getBoundingClientRect().left;
+      const breadcrumbsRight = breadcrumbs.getBoundingClientRect().right;
+      const firstBreadcrumbsLeft = breadcrumbs
+        .querySelector('.breadcrumbs__item:first-child')
+        .getBoundingClientRect().left;
+      const lastBreadcrumbsRight = Math.floor(
+        breadcrumbs
+          .querySelector('.breadcrumbs__item:last-child')
+          .getBoundingClientRect().right,
+      );
+
+      if (firstBreadcrumbsLeft < breadcrumbsLeft) {
```

Comment:
I tried to refactor this, but because there's nested logic, the switch statement was equally spaghetti... and it had some weird bugs (the arrows were showing and hiding in the middle of scrolling). I probably didn't do it right, but it seems that nested switch statements aren't really preferred either.

So, I'm going to leave it as is for now.

Reply:
word 👍